editor/editor_node.cpp

```diff
@@ -1241,9 +1241,10 @@ void EditorNode::_save_scene_with_preview(String p_file, int p_idx) {
 			// The 3D editor may be disabled as a feature, but scenes can still be opened.
 			// This check prevents the preview from regenerating in case those scenes are then saved.
 			Ref<EditorFeatureProfile> profile = feature_profile_manager->get_current_profile();
-			if (!profile->is_feature_disabled(EditorFeatureProfile::FEATURE_3D)) {
-				img = Node3DEditor::get_singleton()->get_editor_viewport(0)->get_viewport_node()->get_texture()->get_data();
-			}
+			if (profile.is_valid())
+				if (!profile->is_feature_disabled(EditorFeatureProfile::FEATURE_3D)) {
```

Comment:
Looks good to me. I am not 100% sure what the current style guidelines are, are you sure that you should create a new if statement and that you shouldn't use curly brackets?

Reply:
Mhhh I base myself with clang-format

Reply:
I didn't find any specifics on if statements in the style guide, but it does refer to 1-2 indentation levels. So maybe consider merging the two ifs to keep the indentation lower.

Reply:
> I didn't find any specifics on if statements in the style guide, but it does refer to 1-2 indentation levels. So maybe consider merging the two ifs to keep the indentation lower.

is_feature_disabled() will crash

Reply:
I am very very very certain c++ supports short-circuiting... Am I missing something?

Reply:
Oh right, i never use them to be more clear. But I could use them if it's required